src/fsi_benchmark_main.ml

```diff
@@ -77,35 +78,42 @@ let string_of_metric = function
   | `with_FP -> "FP"
 
 let print formatter tool result print_metrics : unit =
-  let tool_name = if tool = "bap-byteweight" then "BW" else "IDA" in
+  let tool_name = if tool = "byteweight" then "BW" else "IDA" in
   fprintf formatter "Tool";
   List.iter print_metrics ~f:(fun m -> fprintf formatter "\t%s"
                                @@ string_of_metric m);
   fprintf formatter "\n%s" tool_name;
   output_metric_value formatter result print_metrics
 
 let compare_against bin tool_name truth_name print_metrics : unit =
-  let open Or_error in
-  (Func_start.of_tool tool_name ~testbin:bin >>| fun tool ->
-   Func_start.of_truth truth_name ~testbin:bin >>| fun truth ->
-   let result =
-     let to_set seq = Seq.fold seq ~init:Addr.Set.empty
-         ~f:Addr.Set.add in
-     let tool = to_set tool in
-     let truth = to_set truth in
-     let false_positive = Set.(length (diff tool truth)) in
-     let false_negative = Set.(length (diff truth tool)) in
-     let true_positive = Set.length truth - false_negative in
-     let ratio x = Float.(of_int true_positive / (of_int true_positive + of_int x)) in
-     let prec = ratio false_positive in
-     let recl = ratio false_negative in
-     let f_05 = 1.5 *. prec *. recl /. (0.5 *. prec +. recl) in
-     {false_positive;false_negative;true_positive;prec;recl;f_05} in
-   print std_formatter tool_name result print_metrics) |> function
-  | Ok _ -> ()
-  | Error err ->
-    printf "Function start is not recognized properly due to
-  the following error:\n %a" Error.pp err
+  let rooters = Rooter.Factory.list () in
+  let valid_tool = List.mem rooters tool_name in
+  if not valid_tool
+  then printf "Error: Cannot use tool %S.\n" tool_name
```

Comment:
A couple of issues here:

1. It is better to output diagnostic information into `stderr`, not `stdout`; 
2. Even better is to use `error|warning|...` functions for the diagnostics;
3. If it is possible, the user interaction aspect should be decoupled with the business logic aspect, 
    that can be achieved by throwing an exception on the check failure, and capturing in a place that 
    is closer to a user and that is responsible for the interacting with the user.
4. Even better is to check inputs at the place, where they are introduced to the system, and write functions, that has preconditions that shouldn't bother themselves with checking the precondition.